### ISSUE TITLE
Feature request quickmark (1/3): have the quickmark with 100% label match at the top

### ISSUE THREAD

**Issue by prosoitos** [qutebrowser/qutebrowser#3877]
I love quickmarks. Thanks for this amazing feature :slightly_smiling_face: And I defined many very short labels for fast calling. Because they are short (e.g. "sc"), typing them makes a very long list of quickmarks show up (that's normal and good). But because there is no option to bring the quickmark that would be selected if `Enter` were pressed to the top, it is lost in the list, sometimes very far down. So when I am confident that the page I want to open really has a label "sc", I can blindly press enter and it works. But if I don't remember if it was "sc" or "scl", I have to scroll down a lot to get to the quickmark I am interested in (or give it a blind try and press `Enter` to have DuckDuckGo open with a search of "sc").

I realize that quickmarks are not ordered randomly and there is talk on how to order them (e.g. #3688 and I like the idea of the reverse order by the way). But this could be independent of the ordering: whenever the expression typed matches a label, that quickmark would pop to the top.

In the example I gave, if my quickmark is "scl", it would then be instant to see that typing "sc" brings no quickmark with an "sc" label and that it is not "sc". It would be easy to try a few options to bring the good one up and have my memory refreshed on what it is.

**Comment by jgkamat** [comment]
This can be reproduced by having shorter entries added "older" (or later in the file) than newer entries.

For example:
```
t1 https://github.com/qutebrowser/qutebrowser/issues/3877
t https://github.com/qutebrowser/qutebrowser/issues/3877
```

seems to prefer t1 over t when `:quickmark-load t`.

**Comment by prosoitos** [comment]
Exactly. Sorry for not producing an example (I didn't think of it). But if you press `:open t', `enter`, it will open the correct page. But if `t` is 200 lines down the drop down menu because everything else starting by `t` is ahead, it is hard to make sure it is indeed the right page that will open when pressing `enter`.
